R/object_name_linters.R

```diff
@@ -330,13 +188,38 @@ regexes_rd <- toString(paste0("\\sQuote{", names(style_regexes), "}"))
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 object_length_linter <- function(length = 30L) {
-  object_linter_factory(function(source_file, token) {
-    if (nchar(token$text) > length) {
-        object_lint(
-          source_file,
-          token,
-          paste0("Variable and function names should not be longer than ", length, " characters.")
-        )
-      }
+  lint_msg <- paste0("Variable and function names should not be longer than ", length, " characters.")
+
+  Linter(function(source_file) {
+    if (is.null(source_file$full_xml_parsed_content)) return(list())
+
+    xml <- source_file$full_xml_parsed_content
+
+    assignments <- xml2::xml_find_all(xml, object_name_xpath)
+
+    # Retrieve assigned name
+    nms <- strip_names(
+      xml2::xml_text(assignments)
+    )
+
+    generics <- strip_names(c(
+      declared_s3_generics(xml),
+      imported_s3_generics(namespace_imports(find_package(source_file$filename)))$fun,
```

Comment:
make a wrapper for this? too much nesting --> not clear

Reply:
factored out `ns_imports` as a variable. No need for wrapping IMO, the logic isn't very complex, it's just a chain of functions.